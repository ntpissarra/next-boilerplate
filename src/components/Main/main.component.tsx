import * as S from './main.styles';

const Main = () => {
  return (
    <S.Wrapper>
      <S.Title>Next Boilerplate</S.Title>
      <S.Description>
        TypeScript, ReactJS, NextJS e Styled Components
      </S.Description>
    </S.Wrapper>
  );
};

export default Main;
