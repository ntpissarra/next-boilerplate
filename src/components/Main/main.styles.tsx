import styled from 'styled-components';

export const Wrapper = styled.main`
  background-color: #06092b;
  color: #fff;
  width: 100%;
  height: 100%;
  text-align: center;
  display: flex;
  flex-direction: column;
  justify-content: center;
`;

export const Title = styled.h1`
  padding: 20px;
  font-size: 6rem;
`;

export const Description = styled.p`
  font-size: 3rem;
`;
