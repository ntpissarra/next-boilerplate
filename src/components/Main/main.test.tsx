import { render, screen } from '@testing-library/react';

import Main from './main.component';

describe('<Main />', () => {
  it('should render the main component', () => {
    const { container } = render(<Main />);

    expect(
      screen.getByRole('heading', { name: /Next Boilerplate/i })
    ).toBeInTheDocument();

    expect(container.firstChild).toMatchSnapshot();
  });

  it('should render colors correctly', () => {
    const { container } = render(<Main />);

    expect(container.firstChild).toHaveStyle({ 'background-color': '#06092b' });
  });
});
