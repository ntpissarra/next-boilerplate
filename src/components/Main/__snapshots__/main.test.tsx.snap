// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Main /> should render the main component 1`] = `
.c0 {
  background-color: #06092b;
  color: #fff;
  width: 100%;
  height: 100%;
  text-align: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c1 {
  padding: 20px;
  font-size: 6rem;
}

.c2 {
  font-size: 3rem;
}

<main
  class="c0"
>
  <h1
    class="c1"
  >
    Next Boilerplate
  </h1>
  <p
    class="c2"
  >
    TypeScript, ReactJS, NextJS e Styled Components
  </p>
</main>
`;
